### PR TITLE
Account Recovery: Render different forms based on the state

### PR DESCRIPTION
**Pull request by southp**
## Summary
This PR is derived from the discussion in #12588 , attempting to implement the foundation of chaining the whole AR flow without relying on `page.js`.

In general, this PR includes the following pieces:

1. Leave only two routes in the controller: `/account-recovery/`, `/account-recovery/forgot-username`, since they serve as different entry points.
1. Add `<AccountRecoveryRoot />` component, which encapsulates the logic of showing different form component based on the data at hand.
1. Remove several "page" components, e.g. `<LostPasswordPage />`, `<ForgotUsernamePage />`, since `<AccountRecoveryRoot />` is now the unified page in replacing of them.
1. Make some form elements to be controlled by react component state. These were used to be controlled by redux state directly. However, now that we are showing different forms based on the redux state, it is clearer to only update the redux state after a succeeded form submission.
1. Wrap the form components, e.g. `<FormButton />`, `<FormTextInput />`, in a `<form>` tag and mark the primary button as `type="submit"`, so that a user can simply hit the enter key to submit the form.
1. The data layer handler for `ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST` is now dispatching two actions on success: `fetchResetOptionsSuccess` and `updatePasswordResetUserData`. That is, the `userData` redux state will be populated only after a successful request for the reset options.

## Test Plan

1. Open an incognito window and make sure you are logged out.
1. Go to http://calypso.localhost:3000/account-recovery/
1. Enter a valid, non-a11n account, and hit "continue". You should be directed to "/account-recovery/reset-password"
1. Go to http://calypso.localhost:3000/account-recovery/ again.
1. Hit "Forgot username?". You should be directed to "/account-recovery/forgot-username"
1. Enter valid personal profile of an non-a11n account, and hit "continue". You should be directed to "/account-recovery/reset-password".

### CODE REVIEW

**Comment by mattwondra** [comment]
@southp Is this supposed to trigger a "success" screen of some sort after you choose how to reset the password? When I press "Continue" on that screen I get the recovery email/text, but the form stays up and it looks like nothing happened.

**Comment by southp** [comment]
@mattwondra Oh, it is intentional :) This PR only chains the flow from step 2 ~ 4 to build the foundation. The complete flow chart is here: https://cloudup.com/cC9EfchsqDg.

File: client/account-recovery/lost-password-form/index.jsx [hunk 1]
```diff
@@ -79,30 +84,32 @@ export class LostPasswordFormComponent extends Component {
...
+							<FormInput
+								className="lost-password-form__user-login-input"
+								onChange={ this.onUserLoginChanged }
+								value={ userLoginFormValue }
+								disabled={ isRequesting } />
+						</FormLabel>
+						{
+							requestError && (
+							<p className="lost-password-form__error-message">
+								{ translate( 'We encountered some problems with that login information. ' +
```
Comment (jordwest) [inline client/account-recovery/lost-password-form/index.jsx]: I saw this error when I typed an incorrect email address but found the message a bit confusing, as if there was an error out of my control. Could we be more specific here?

Comment (southp) [inline client/account-recovery/lost-password-form/index.jsx]: Actually, I can't recall the original purpose here. I think it was just me dumping whatever came to my mind at the moment. I will discuss it with @evilluendas and open another PR soliciting copy review for this.

**Comment by southp** [comment]
> The only thing that would be nice to add is a way to navigate back to the previous screen
Oh, that's a good idea. I will implement in a separate PR to prevent further bloating this PR :)

**Pull request merged**